docs/data/sdks/typescript-browser/index.md

```diff
@@ -340,6 +340,17 @@ amplitude.setGroup('orgId', '15');
 amplitude.setGroup('sport', ['soccer', 'tennis']);
 ```
 
+--8<-- "includes/event-level-groups-intro.md"
+
+```ts
+amplitude.track({
+    event_type: 'event type',
+    groups: { 'orgId': '15' }
+},
+    { eventPropertyKey: 'event property value' }
+);
```

Comment:
Event properties should be part of the event payload. See other suggestion above.

Reply:
Typescript client accepts `eventProperties` as an optional parameter as well: https://github.com/amplitude/Amplitude-TypeScript/blob/6262330785394f88ffbd1ed1aaa72f48e7d50f3e/packages/analytics-core/src/core-client.ts#L55

I can see why you want me to put event properties as part of base event is to make it align with other SDKs. But curious why do we event support passing event properties as an optional parameter in `track`

Reply:
> why do we event support passing event properties as an optional parameter in track

Because the type of first parameter of TS SDKs is `string | BaseEvent`. So this is TS SDKs specific to support passing only event type.